tests/Functional/LimitFunctionalTest.php

```diff
@@ -0,0 +1,73 @@
+<?php
+
+namespace Keboola\S3ExtractorTest\Functional;
+
+class LimitFunctionalTest extends FunctionalTestCase
+{
+    public function testLimitReached(): void
+    {
+        $this->runTestWithCustomConfiguration(
+            __DIR__ . '/limit/reached',
+            [
+                'parameters' => [
+                    'accessKeyId' => getenv(self::AWS_S3_ACCESS_KEY_ENV),
+                    '#secretAccessKey' => getenv(self::AWS_S3_SECRET_KEY_ENV),
+                    'bucket' => getenv(self::AWS_S3_BUCKET_ENV),
+                    'key' => 'f*',
+                    'includeSubfolders' => true,
+                    'newFilesOnly' => false,
+                    'limit' => 1,
+                ],
+            ],
+            0,
+            null,
```

Comment:
V původním testu se tady ještě kontrolovaly výstupy na `Monolog\Handler\TestHandler`.

https://github.com/keboola/aws-s3-extractor/blob/397cc75f6b6b2afae681f5acef7e90dfd5af17fd/tests/Keboola/S3ExtractorTest/Functional/LimitFunctionalTest.php#L44-L45

Log handler teď nejde injectnout, ale `runTestWithCustomConfiguration` umí přes argument vložit `$expectedStdout`. Výstupy bych určitě checkoval. Bude to pravěpodobně chybět u víc testů.

Reply:
good point, tohle není nikde, už to doplňuji 👍

Reply:
done https://github.com/keboola/aws-s3-extractor/pull/50/commits/7aad19a19ea545768506569c35683cf86bf35127